### PR TITLE
chore: improve kafka startup logging

### PR DESCRIPTION
**Pull request by lzap**
It looks like we use TLS on stage but not on prod and ephemeral.

This is small logging improvement that should confirm it as we do not dump configuration on prod.

### CODE REVIEW

File: internal/kafka/kafka.go [hunk 1]
```diff
@@ -98,7 +106,6 @@ func NewKafkaBroker(ctx context.Context) (Broker, error) {
 
 		//nolint:gosec
 		tlsConfig = &tls.Config{
-			MinVersion:         tls.VersionTLS12,
```
Comment (lzap) [inline internal/kafka/kafka.go:101]: Unrelated, TLS 1.2 is the default value when this field is `nil` so useless.

File: internal/kafka/kafka.go [hunk 2]
```diff
@@ -81,7 +81,15 @@ func NewKafkaBroker(ctx context.Context) (Broker, error) {
 	var saslMechanism sasl.Mechanism
 
 	logger := zerolog.Ctx(ctx)
-	logger.Debug().Msgf("Setting up Kafka transport: %v", config.Kafka.Brokers)
+	logger.Info().Msgf("Setting up Kafka transport: %v TLS:%b Skip:%b CA:%b Auth:%s SASLMech:%s SASLProto:%s",
```
Comment (ezr-ondrej) [inline internal/kafka/kafka.go:84]: I guess %b should have been %t?
```suggestion
	logger.Info().Msgf("Setting up Kafka transport: %v TLS:%t Skip:%t CA:%t Auth:%s SASLMech:%s SASLProto:%s",
```

File: internal/kafka/kafka.go [hunk 3]
```diff
@@ -81,7 +81,15 @@ func NewKafkaBroker(ctx context.Context) (Broker, error) {
...
 	logger := zerolog.Ctx(ctx)
-	logger.Debug().Msgf("Setting up Kafka transport: %v", config.Kafka.Brokers)
+	logger.Info().Msgf("Setting up Kafka transport: %v TLS:%t Skip:%t CA:%t Auth:%s SASLMech:%s SASLProto:%s",
+		config.Kafka.Brokers,
+		config.Kafka.TlsEnabled,
+		config.Kafka.TlsSkipVerify,
+		config.Kafka.CACert != "",
+		config.Kafka.AuthType,
+		config.Kafka.SASL.SaslMechanism,
+		config.Kafka.SASL.SecurityProtocol,
```
Comment (ezr-ondrej) [inline internal/kafka/kafka.go:91]: aren't we more interested in what's in the clowder config directly?

**Review by ezr-ondrej — APPROVED** [review]
sure, let's get it in :)

**Comment by lzap** [comment]
Do not merge, I have some new info about clowder.

**Comment by lzap** [comment]
After chat with Derek Horton, he suggest to provide an empty TLS config value as the library is supposed to automatically detect if TLS is present on a cluster or not and use the safest transport. He mentioned that he uses this on their project and it works even after Kafka migration from this weekend.

Therefore this patch drops the newly introduced setting completely, I don’t have a TLS Kafka setup in my home lab so we need to test this. Ephemeral is plaintext and stage is TLS so if both works prod should be fine. Then I will remove these flags.

**Comment by lzap** [comment]
**DO NOT MERGE.**

Looks like Derek was not right, might not work as he described.

**Comment by lzap** [comment]
I am reverting the commit back to the initial one - only improves logging and drops a line that has no effect. Looks like our solution is the best for now. I asked via email if we can use one of clowder's fields to detect TLS, let’s wait until I get some reply.

**Comment by lzap** [comment]
_Regarding how apps should know whether SSL is on or off moving forward... I think a good example we have is the one Dan Kuc just pushed for malware-detection[1]. You are correct that the only values Clowder sets on this field is either "SSL"[2] or "SASL_SSL"[3,4]. In short, I think you can use this logic to determine if your Kafka client should use SSL or not:

* if kafka.brokers[0].securityProtocol is defined and its value is "SSL" or "SASL_SSL", then SSL should be ON

Your client should be able to use SSL with or without the 'cacert' field present. If the field is present, then use that as your client's trust store. If not, then ideally your client falls back to using the "system default trust store" -- which should already trust well-known root CA's like Digicert, Verisign, LetsEncrypt, and so on._

So I will go ahead and drop our own field and rely on this field instead.

**Comment by lzap** [comment]
So the latest version drops the KAFKA_TLS_ENABLED flag and uses SecurityProtocol flag to check if it contains "SSL" string. If it does, it configures TLS.

The same way it also checks for SASL in the AuthType which should contain this string if SASL should be used.

I am keeping the verification skip flag tho, can be useful for local TLS testing. Once this is confirmed to work on both stage and ephemeral we can remove flags from app-interface. I believe this is the final version.

File: internal/config/config.go [hunk 4]
```diff
@@ -155,17 +155,16 @@ var config struct {
...
-		TlsSkipVerify bool     `env:"TLS_SKIP_VERIFY" env-default:"false" env-description:"do not verify TLS server certificate"`
-		Brokers       []string `env:"BROKERS" env-default:"localhost:9092" env-description:"kafka hostname:port list of brokers"`
-		AuthType      string   `env:"AUTH_TYPE" env-default:"" env-description:"kafka authentication type (mtls, sasl or empty)"`
-		CACert        string   `env:"CA_CERT" env-default:"" env-description:"kafka TLS CA certificate path (use the OS cert store when blank)"`
-		SASL          struct {
-			Username         string `env:"USERNAME" env-default:"" env-description:"kafka SASL username"`
-			Password         string `env:"PASSWORD" env-default:"" env-description:"kafka SASL password"`
-			SaslMechanism    string `env:"MECHANISM" env-default:"" env-description:"kafka SASL mechanism (scram-sha-512, scram-sha-256 or plain)"`
-			SecurityProtocol string `env:"PROTOCOL" env-default:"" env-description:"kafka SASL security protocol"`
+		Enabled          bool     `env:"ENABLED" env-default:"false" env-description:"kafka service enabled"`
```
Comment (ezr-ondrej) [inline internal/config/config.go:158]: should we delete it right away, given we are  now quite sure it's probably not gonna be used? :)

File: internal/kafka/kafka.go [hunk 5]
```diff
@@ -93,18 +100,17 @@ func NewKafkaBroker(ctx context.Context) (Broker, error) {
 		}
 	}
 
-	if config.Kafka.TlsEnabled && !config.InEphemeralClowder() {
+	if strings.Contains(strings.ToUpper(config.Kafka.SecurityProtocol), "SSL") {
```
Comment (ezr-ondrej) [inline internal/kafka/kafka.go:103]: I'd love to have the values in constants, but I guess we can contribute that to https://github.com/RedHatInsights/app-common-go and then use it.
So good for now :+1:

**Review by ezr-ondrej — COMMENTED** [review]
Just two nits and only one of them addressable xD
Apart of it great effort here to find out how to do it properly! :+1:

File: internal/config/config.go (hunk 4 above)
Comment (lzap) [reply internal/config/config.go:158]: Delete what, enabled flag? How are you going to test Kafka in dev environment (without clowder)?

File: internal/kafka/kafka.go (hunk 5 above)
Comment (lzap) [reply internal/kafka/kafka.go:103]: Ehm "middleware" ehm.

**Comment by lzap** [comment]
So we got the following info for the field: _PLAINTEXT, SSL, SASL_PLAINTEXT, or SASL_SSL_ I can incorporate this in the help text if necessary.

File: internal/kafka/kafka.go (hunk 5 above)
Comment (ezr-ondrej) [reply internal/kafka/kafka.go:103]: right xD :)

File: internal/config/config.go (hunk 4 above)
Comment (lzap) [reply internal/config/config.go:158]: Still not sure what you mean here, rebasing anyeway.

**Comment by lzap** [comment]
Rebased the docs-only change.

**Comment by lzap** [comment]
Fails on `unexpected backend data` that is the error we talked about on scrum. Something is deleting source prior this request. @akhil-jha any ideas?

**Comment by ezr-ondrej** [comment]
Thanks! :orange_heart:

**Pull request merged**